[BUGFIX] Fix CharSelect not playing confirm animation mid-transition

Lasercar opened the pull request:
<!-- Please read the Contributing Guide (https://github.com/FunkinCrew/Funkin/blob/main/docs/CONTRIBUTING.md) before submitting this PR. -->
## Does this PR close any issues? If so, link them below.
Fixes #3730
## Briefly describe the issue(s) fixed.
Characters do not play confirm animation if selected mid transition. The pressed Selected bool is now passed when they're switched and plays the confirm animations instead if it's true.

## Include any relevant screenshots or videos.

[See trayfellow's video below](https://github.com/FunkinCrew/Funkin/pull/4272#issuecomment-4122867719)

Lasercar committed:
Character Select fix
Characters do not play confirm animation if selected mid transition

Lasercar committed:
Character Select fix
Characters do not play confirm animation if selected mid transition

trayfellow commented:
Lasercar carrying
![ezgif-76b21c61936a5c](https://github.com/user-attachments/assets/a8ce37ac-cac1-4df1-b2e8-adb45bdc0df4)

Starexify commented:
Could this also fix the memory leak from spam selecting between characters? i got to around 10gb ram usage only from doing that (yes I am sane)

Lasercar commented:
> Could this also fix the memory leak from spam selecting between characters? i got to around 10gb ram usage only from doing that (yes I am sane)

Sure, I'll see if I can do a simple cache hack.

Hundrec commented:
Not familiar with the bugs this PR fixes, but it needs a conflict resolution

Hundrec reviewed (changes requested):
This almost works right, but sometimes I can catch Boyfriend or Pico not doing their confirm anyway.
I was pressing left/right and Enter at nearly the same time to get it to happen

Hundrec reviewed (changes requested):
Wait, I just tested it a little more and got both characters to freeze during the fade out transition to Freeplay.

I think this is accidentally freezing them, not sure if Escape is involved. (I think it's when you switch/select really quickly)

Trayfellow committed:
Improve upon Lasercar's CS fixes (see details)

Improvements for `CS not playing confirm animation mid-transition`

- (fix) Confirm anim via idle skip for GF now correctly loops instead of playing once

- (tweak) Player now skips waiting for slide transition to finish, just play the select animation instantly instead

trayfellow committed:
Improve upon Lasercar's CS fixes (see details)

Improvements for `CS not playing confirm animation mid-transition`

- (fix) Confirm anim via idle skip for GF now correctly loops instead of playing once

- (tweak) Player now skips waiting for slide transition to finish, just play the select animation instantly instead

Hundrec commented:
Abnormal fixed this internally, so this change is no longer needed. 
Thank you for your contribution!

Lasercar commented:
got it
